### PR TITLE
[master] intraprocess delivery [6930]

### PR DESCRIPTION
**Pull request by IkerLuengo**
This is a port of #854 + #923 + 92efb9257bfc8e725558af646ff96040b461c995

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/989/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/817/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/985/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/294/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/995/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/824/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/991/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/300/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1003/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/834/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/999/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/308/)

**Comment by IkerLuengo** [comment]
Relaunched windows tests:
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1031/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1048/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/870/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1046/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/341/)

**Comment by IkerLuengo** [comment]
Failure in linux is corrected in #893  (merge pending)

**Comment by IkerLuengo** [comment]
#893 merged. Linux job should not fail anymore

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1054/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/875/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1052/)
* Performance [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Performance%20Tests%20Localhost/347/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1060/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/881/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1058/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1139/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/954/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1139/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1157/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/964/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1156/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1176/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/975/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1174/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1193/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/988/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1191/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1219/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1006/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1216/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1283/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1068/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1280/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1320/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1101/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1317/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1347/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1120/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1344/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1430/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1203/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1428/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1433/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1206/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1431/)

File: include/fastrtps/attributes/LibrarySettingsAttributes.h [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    bool operator==(
+            const LibrarySettingsAttributes& b) const
+    {
+        return (intraprocess_delivery == b.intraprocess_delivery);
+    }
+
+    IntraprocessDeliveryType intraprocess_delivery = INTRAPROCESS_FULL;
+};
+
+}
```
Comment (MiguelCompany) [inline include/fastrtps/attributes/LibrarySettingsAttributes.h]:
```suggestion
}  // namespace fastrtps
```

File: include/fastrtps/attributes/LibrarySettingsAttributes.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
```
Comment (MiguelCompany) [inline include/fastrtps/attributes/LibrarySettingsAttributes.h]:
```suggestion
}  // namespace eprosima
```

File: src/cpp/rtps/xmlparser/XMLElementParser.cpp [hunk 3]
```diff
@@ -2900,6 +2900,53 @@ XMLP_ret XMLParser::getXMLBool(
     return XMLP_ret::XML_OK;
 }
 
+XMLP_ret XMLParser::getXMLEnum(
+        tinyxml2::XMLElement *elem,
```
Comment (MiguelCompany) [inline src/cpp/rtps/xmlparser/XMLElementParser.cpp]:
```suggestion
        tinyxml2::XMLElement* elem,
```

File: src/cpp/rtps/xmlparser/XMLElementParser.cpp [hunk 4]
```diff
@@ -2900,6 +2900,53 @@ XMLP_ret XMLParser::getXMLBool(
...
+        IntraprocessDeliveryType * e,
```
Comment (MiguelCompany) [inline src/cpp/rtps/xmlparser/XMLElementParser.cpp]:
```suggestion
        IntraprocessDeliveryType* e,
```

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/1448/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/1221/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/1446/)

File: src/cpp/rtps/xmlparser/XMLElementParser.cpp [hunk 5]
```diff
@@ -2900,6 +2900,53 @@ XMLP_ret XMLParser::getXMLBool(
...
+    }
+    else if (strcmp(text, OFF) == 0)
+    {
+        *e = IntraprocessDeliveryType::INTRAPROCESS_OFF;
+    }
+    else if (strcmp(text, USER_DATA_ONLY) == 0)
+    {
+        *e = IntraprocessDeliveryType::INTRAPROCESS_USER_DATA_ONLY;
+    }
+    else if (strcmp(text, FULL) == 0)
```
Comment (MiguelBarro) [inline src/cpp/rtps/xmlparser/XMLElementParser.cpp]: We must be aware that:
```xml
          <library_settings>
              <intraprocess_delivery>
                  OFF
              </intraprocess_delivery>
          </library_settings>
```
neither maches the schema nor can be parsed by the new getXMLEnum override.

**Pull request merged**